rest/data/project_settings.go

```diff
@@ -56,6 +56,15 @@ func CopyProject(ctx context.Context, env evergreen.Environment, opts CopyProjec
 	disableStartingSettings(projectToCopy)
 
 	u := gimlet.GetUser(ctx).(*user.DBUser)
+	isAdmin := u.HasPermission(gimlet.PermissionOpts{
```

Comment:
We can't / shouldn't add this here -- we use middleware to handle permission checking before we ever get to the route logic, which is why the route currently doesn't use the createProject logic and instead uses the requireProjectAdmin logic (which should be fine to keep). If the idea is to fix this for Spruce, it may be to change the directive used here https://github.com/evergreen-ci/evergreen/blob/adacf7967c2df3c5b4db9f551b02efb3ce983e82/graphql/schema/mutation.graphql#L52 . I think it would be worth splitting that work to a separate ticket.